starlette/testclient.py

```diff
@@ -22,34 +23,40 @@ class _MockOriginalResponse(object):
     it was made using urllib3.
     """
 
-    def __init__(self, headers):
+    def __init__(self, headers: BytesPairs) -> None:
         self.msg = _HeaderDict(headers)
         self.closed = False
 
-    def isclosed(self):
+    def isclosed(self) -> bool:
         return self.closed
 
 
 class _Upgrade(Exception):
-    def __init__(self, session):
+    def __init__(self, session: "WebSocketTestSession") -> None:
         self.session = session
 
 
-def _get_reason_phrase(status_code):
+def _get_reason_phrase(status_code: int) -> str:
     try:
         return http.HTTPStatus(status_code).phrase
     except ValueError:
         return ""
 
 
 class _ASGIAdapter(requests.adapters.HTTPAdapter):
-    def __init__(self, app: typing.Callable, raise_server_exceptions=True) -> None:
+    def __init__(self, app: ASGIApp, raise_server_exceptions: bool = True) -> None:
         self.app = app
         self.raise_server_exceptions = raise_server_exceptions
 
-    def send(self, request, *args, **kwargs):
-        scheme, netloc, path, params, query, fragement = urlparse(request.url)
+    def send(  # type: ignore
+        self, request: requests.PreparedRequest, *args: typing.Any, **kwargs: typing.Any
+    ) -> requests.Response:
+        scheme, netloc, path, params, query, fragement = urlparse(  # type: ignore
+            request.url
+        )
+
         if ":" in netloc:
+            port: typing.Any
             host, port = netloc.split(":", 1)
```

Comment:
Let's use `host, port_string` here, then `port = int(port_string)` in order to avoid the `Any` annotation.